src/libasr/codegen/wasm_decoder.h

```diff
@@ -262,8 +262,13 @@ class WASMDecoder {
             }
 
             data_segments.p[i].insts_start_index = offset;
-            while (read_b8(wasm_bytes, offset) != 0x0B)
```

Comment:
For the following example shared at #1453 
```python
def main0():
    print("01234567891") # string of length 11
    print("hi")

main0()
```
The crux of the issue is that strings length `11` when encoded using `leb128` signed integer is `0x0B` which is also the `byte` for expression/instructions end in `wasm`.

So, the above `while` loop terminates at the first `0x0B` encountered (which is of `11`'s encoding and does not represent expression/instructions end). As the loop terminates early, the remaining bytes of `wasm_bytes` are not meaningful and thus leads to unexpected errors like in https://github.com/lcompilers/lpython/issues/1453#issue-1553212840 and https://github.com/lcompilers/lpython/issues/1453#issuecomment-1400471921.